src/tests/encore/parametric/methodPolyClosure.enc

```diff
@@ -0,0 +1,18 @@
+read class Bar[t]
+  def bar() : unit
+    println("In Bar")
+  end
+end
+
+read class Foo
+  def foo[ty]() : () -> unit
+    fun() => (new Bar[ty]()).bar()
```

Comment:
This works for this case but doesn't work for the following one (copy-paste of this function plus a local function declaration):

```read class Bar[t]
  def bar() : unit
    println("In Bar")
  end
end

read class Foo
  def foo[ty]() : () -> unit
    fun() => (new Bar[ty]()).bar()
  where
    fun foo(): unit
      new Bar[ty]()
    end
  end
end

active class Main
  def main() : unit
    val f = (new Foo()).foo[int]()
    f()
  end
end
```